File: Buckeye Code/Other/rpncalculator.py
standard_input = '''8
2
enter
4
enter
6
*
swap
-
'''

n = int(input())
stack = []
for i in range(n):
    stack.append(0)
for i in range(n):
    op = input()
    # print(stack, op)
    if op == '+':
        stack.append(stack.pop(-1) + stack.pop(-1))
    elif op =='-':
        top = stack.pop(-1)
        bot = stack.pop(-1)
        stack.append(bot - top)
    elif op == '*':
        stack.append(stack.pop(-1) * stack.pop(-1))
    elif op == '/':
        top = stack.pop(-1)
        bot = stack.pop(-1)
        stack.append(bot // top)
    elif op == 'swap':
        top = stack.pop(-1)
        bot = stack.pop(-1)
        stack.append(top)
        stack.append(bot)
    elif op == 'enter':
        stack.append(stack[-1])
    else:
        num = int(op)
        if len(stack) == 0:
            stack.append(num)
        else:
            stack[-1] = num
    
print(stack[-1])

File: Buckeye_Code/rpncalculator.java
import java.util.Comparator;
import java.util.Scanner;
import java.util.Stack;

public class rpncalculator {

    // Comparator Template

    /**
     * Compare {@code String}s in alphabetical order
     */
    private static class StringLT implements Comparator<String> {
        @Override
        public int compare(String o1, String o2) {
            return o1.compareToIgnoreCase(o2);
        }
    }

    // Private Static Methods Starts Here

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // Code Starts Here
        int n = Integer.parseInt(sc.nextLine());
        Stack<Integer> stack = new Stack<>();
        stack.push(0);
        for (int i = 0; i < n; i++) {
            String op = sc.nextLine();
            if (op.equals("+")) {
                stack.push(stack.pop() + stack.pop());
            } else if (op.equals("-")) {
            	int num = stack.pop();
                int num2 = stack.pop();
                stack.push(num2 - num);
            } else if (op.equals("*")) {
                stack.push(stack.pop() * stack.pop());
            } else if (op.equals("/")) {
                int num = stack.pop();
                int num2 = stack.pop();
                stack.push(num2 / num);
            } else if (op.equals("enter")) {
                stack.push(stack.peek());
            } else if (op.equals("swap")) {
                int num = stack.pop();
                int num2 = stack.pop();
                stack.push(num);
                stack.push(num2);
            } else {
                if (stack.size() != 0) {
                    stack.pop();
                }
                stack.push(Integer.valueOf(op));
            }
        }
        System.out.println(stack.pop());
    }

}